server/src/main/java/org/opensearch/transport/InboundDecoder.java

```diff
@@ -217,7 +217,12 @@ static IllegalStateException ensureVersionCompatibility(Version remoteVersion, V
         // handshake. This looks odd but it's required to establish the connection correctly we check for real compatibility
         // once the connection is established
         final Version compatibilityVersion = isHandshake ? currentVersion.minimumCompatibilityVersion() : currentVersion;
-        if ((currentVersion.equals(Version.V_2_0_0) && remoteVersion.equals(Version.fromId(6079999))) == false
+
+        // todo: remove in 4x as handshake version spoofing will no longer be needed to ensure mixed cluster compatibility
+        boolean v2x = currentVersion.onOrAfter(Version.V_2_0_0) && currentVersion.before(Version.V_3_0_0);
+        boolean v3x = currentVersion.onOrAfter(Version.V_3_0_0) && currentVersion.before(Version.fromId(4000000));
```

Comment:
Should we introduce `NEXT` constant for that? ( `== Version.fromId(4000000)` for now)

Reply:
:+1: